docs/DeveloperGuide.md

```diff
@@ -377,11 +416,94 @@ Use case ends.
 
 3a. The given index is invalid
 
-> 3a1. AddressBook shows an error message <br>
+> 3a1. myPotato shows an error message <br>
   Use case resumes at step 2
 
+#### Use Case: Sort tasks by date
+
+**MSS**
+
+1. User choose a date;
+2. myPotato show all the tasks in that day;
+3. User type in command sort;
+4. myPotato sort the tasks according to deadlines;
+5. myPotato shows the list of tasks according to deadlines;
+Use case ends.
+
+**Extensions**
+
+2a. If there is no task in that day
+
+> 2a1. myPotato shows an error message <br>
+
+3a.If there are more than one tasks sharing the same deadline:
+
+> 3a1. System show the list according to alphabetic order among those tasks.
+
+#### Use Case: Edit tasks
+
+**MSS**
+
+1. User requests to list tasks
+2. myPotato shows a list of tasks
+3. User requests to change a specific task
+4. myPotato shows the specific task
+5. User makes the changes
+6. myPotato saves and update changes <br>
+Use case ends.
+
+**Extensions**
+
+2a. The list is empty
+
+> Use case ends
+
+3a. The given name is invalid
+
+> 3a1. myPotato shows an error message<br>
+  Use case resumes at Step 2
+ 
+####Use case: Undo a task
+
+**MSS**
+
+1. User requests to show the list of deleted tasks
```

Comment:
there is something wrong with the use case.
- mainly it is not covering all possible undo scenarios
- why would you want to show a list of deleted tasks and then undo? kinda counter intuitive how undo should work